distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ErrorPumpThread.java

```diff
@@ -84,8 +84,20 @@ void drain() {
         nonInterruptibleVoid(this::join);
     }
 
-    /** List of messages / lines to filter from the output. */
-    List<String> filter = List.of("WARNING: Using incubator modules: jdk.incubator.vector");
+    /** Set of messages / lines to filter from the output. */
+    private static Set<String> filter = Set.of("WARNING: Using incubator modules: jdk.incubator.vector");
+
+    private static boolean isFilteredOut(String line) {
+        if (filter.contains(line)) {
```

Comment:
How about making the filter a predicate? then we can still have one set of filters, instead of some in the set and some in here manually